src/main/groovy/jenkins/automation/utils/ScmUtils.groovy

```diff
@@ -57,4 +60,31 @@ class ScmUtils {
             }
         }
     }
+
+/***
+ *
+ * Utility method to do git shallow-clones easily.
+ * @param context A reference to the job object being modified
+ * @param url A string containing a git repo URL
+ * @param branch Optional branch to check-out; defaults to master
+ *
+ */
+    static void shallowGit(context, url, branch = "master") {
+        context.with {
+            git {
+                remote {
+                    delegate.url(url)
+                }
+
+                delegate.branch(branch)
+
+                extensions {
+                    cloneOptions {
+                        shallow()
+                    }
+                }
+            }
+        }
+    }
+
```

Comment:
Please remove extra blank line.

Reply:
@imuchnik That one is intentional -- it matches the blank line on line 10 at the start of the class body. Highly subjective, for sure, but there appears to be a pretty varied mix of whitespace styles across the source in this repo.